libs/bitmap-convert/src/base70.rs

```diff
@@ -0,0 +1,76 @@
+use bitmaps::Bitmap;
+use bnum::types::U256;
+
+use crate::error::Error;
+
+const BASE_70: &str =
+    "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ-~._()!*";
+
+pub trait BitmapBase70Conv {
+    type Error;
+
+    fn to_base_70(&self) -> Result<String, Self::Error>;
+
+    fn base_70_to(string: String) -> Result<Self, Self::Error>
+    where
+        Self: Sized;
+}
+
+impl BitmapBase70Conv for Bitmap<256> {
+    type Error = Error;
+
+    fn to_base_70(&self) -> Result<String, Self::Error> {
+        // 将字符串转换成索引
+        let mut char_to_index: [u8; 127] = [0; 127];
+        let index_to_char: Vec<char> = BASE_70.chars().collect();
+        let radix = index_to_char.len() as u32;
+        for (i, c) in index_to_char.iter().copied().enumerate() {
+            char_to_index[c as usize] = i as u8;
+        }
+
+        // 转换bitmap成u8数组
+        let value = U256::from_radix_le(self.as_bytes(), 256)
+            .ok_or(Error::LargeThen256)?;
+        let bytes = value.to_radix_le(radix);
+        let mut result: Vec<char> = Vec::new();
+        // 转换为70进制的字符数组
+        for b in bytes {
+            result.push(index_to_char[b as usize]);
+        }
+
+        // str是转换进制后的字符串
+        Ok(result.into_iter().collect())
```

Comment:
这边可以直接用迭代器collect 吧
```suggestion
        // 转换为70进制的字符数组
        let s = bytes.into_iter().map(|b|index_to_char[b as usize])
        .collect()
        Ok(v)
```
顺便一提，String 应该可以直接从Vec 转换